### PR TITLE
Improve documentation for new design, continuation of draft PR #49

### PR DESCRIPTION
**Pull request by iuliadmtru**
This is a continuation of my draft PR #49 which comes from a different branch (`docs_improve` instead of `design_update`). I will close PR #49 and continue working on this PR.

### CODE REVIEW

**Comment by iuliadmtru** [comment]
I think I messed up using git.. The code is fine, but the history is all cluttered. Sorry about that! :(

**Comment by iuliadmtru** [comment]
This is a very long PR containing documentation work. It is not completely up to date (or at least it won't be up to date when PR #69 is merged). I think we can manage the conflicts.

File: src/ht.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# TODO: add docstrings
+function HorwitzThompson_HartleyRaoApproximation(y::AbstractVector, design::AbstractSurveyDesign, HT_total)
```
Comment (ayushpatnaikgit) [inline src/ht.jl:2]: Why are these modifications part of this PR?

File: src/svymean.jl [hunk 2]
```diff
@@ -1,49 +1,124 @@
...
-function svymean(x::AbstractVector , design::SimpleRandomSample, _)
-    return DataFrame(mean = mean(x), sem = sem(x, design::SimpleRandomSample))
+function svymean(x::AbstractVector, design::SimpleRandomSample, weights)
+    return DataFrame(mean = mean(x), sem = sem_svyby(x, design::SimpleRandomSample, weights))
 end
+
+
+# StratifiedSample
+
+function svymean(x::Symbol, design::StratifiedSample)
```
Comment (ayushpatnaikgit) [inline src/svymean.jl:94]: This also shouldn't be part of this PR

**Review by ayushpatnaikgit — APPROVED** [review]

File: src/ht.jl (hunk 1 above)
Comment (iuliadmtru) [reply src/ht.jl:2]: These come from my git mess up. I merged this branch with the `design_update` branch at some point in a wrong way.

File: src/svymean.jl (hunk 2 above)
Comment (iuliadmtru) [reply src/svymean.jl:94]: I believe this is the same git mess up.
